Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -5269,12 +5269,46 @@ - (void)changeSelectionWithGestureAt:(CGPoint)point withGesture:(WKBEGestureType
     });
 }
 
+- (void)updateSelection
+{
+    if (_isInSelectionInteraction)
+        [self updateSelectionWithTouchAt:[self convertPoint:_lastSelectionTouch.point fromView:self.webView] withSelectionTouch:WKBESelectionTouchPhaseMoved baseIsStart:_lastSelectionTouch.baseIsStart withFlags:_lastSelectionTouch.flags];
+}
+
 - (void)changeSelectionWithTouchAt:(CGPoint)point withSelectionTouch:(WKBESelectionTouchPhase)touch baseIsStart:(BOOL)baseIsStart withFlags:(WKBESelectionFlags)flags
 {
     logTextInteraction(__PRETTY_FUNCTION__, self.textInteractionLoupeGestureRecognizer, point, std::nullopt, std::nullopt, touch, flags);
 
     _autocorrectionContextNeedsUpdate = YES;
     _usingGestureForSelection = YES;
+    auto touchType = toSelectionTouch(touch);
+    switch (touchType) {
+    case WebKit::SelectionTouch::Started:
+        _isInSelectionInteraction = YES;
+        _lastSelectionTouch.point = [self convertPoint:point toView:self.webView];
+        _lastSelectionTouch.baseIsStart = baseIsStart;
+        _lastSelectionTouch.flags = flags;
+        break;
+    case WebKit::SelectionTouch::Moved:
+        if (_isInSelectionInteraction) {
+            _lastSelectionTouch.point = [self convertPoint:point toView:self.webView];
+            _lastSelectionTouch.baseIsStart = baseIsStart;
```

Comment:
Would you need to update the last cached `_lastSelectionTouch.baseIsStart` here, in the case where the selection flips? (That way, we wouldn't end up sending `_lastSelectionTouch.baseIsStart` with a stale value of `baseIsStart`)

Reply:
Hm... it probably works fine as long as the user moves (at least slightly) after the selection flips, because then `changeSelectionWithTouchAt:withSelectionTouch:baseIsStart:withFlags:` will be called with the updated `baseIsStart`. However, if you keep the selection still after flipping, it's unclear how the last cached `baseIsStart` would be updated to reflect the flipping.

Reply:
I think it would be very difficult to flip and then stop moving, so it would be really difficult to get into that state, but I'll fix it :)